### DISCUSSION TITLE
ZINC dataset data dictionary

### DISCUSSION THREAD

**Discussion by guo00413** [pyg-team/pytorch_geometric#3451] [Q&A]
Hello, I have questions regarding the data dictionary for ZINC dataset.

1. I found this [README.txt](https://github.com/pyg-team/pytorch_geometric/files/7494194/README.txt) from [this site](https://chrsmrrs.github.io/datasets/docs/datasets/) referred by [graph deep learning benchmark](https://github.com/graphdeeplearning/benchmarking-gnns). So, first to confirm, are these labels of nodes and edges also correct for ZINC dataset downloaded from `pytorch_geometric`?
2. After fitting a GNN using this data, I want to apply it with other data coming as SMILE format. There are some chiral type defined in SMILE, e.g. [C@H] or [C@@]. Maybe I missed anything but how does this represented in ZINC dataset loaded from `pytorch_geometric`?
3. Similar to above question, in SMILE, people use `/` and `\` to describe direction of single bonds around a double bond (stereoisomers). Do we know how does this represented in the graphic dataset? I found we only have 3 types of the edges.

I am from applied math background so those questions might be silly if you have a strong chemistry background. Really appreciate your help. Thank you so much.

**Comment by rusty1s** [comment]
Sadly, I'm no expert in chemistry either :)

1. Yes, those are the same datasets.
2. It's probably best to look at the original SMILES data for this, see [here](https://github.com/wengong-jin/icml18-jtnn/tree/master/data/zinc). The repo also contains the code that was used to convert SMILES into graphs.
3. I'm pretty sure directions are ignored, and each graph is undirected.

**Reply by guo00413** [reply]
@rusty1s thanks a lot for the direction. I guess I will use SMILES data and convert the SMILES to graph following this codes instead of ZINC dataset in `pytorch_geometric`.

1. Got it!
2. In their work, they included more features in nodes than what is included in ZINC dataset provided in `pytorch_geometric`, which includes chiral tag, explicit H nums and formal charge.
3. In the repo you referred, they actually considered stereoisomers as edge feature (instead of edge direction, so the graph is still undirected), e.g. please see https://github.com/wengong-jin/icml18-jtnn/blob/28ed03fcb3f0a79f44f73eefc0bcad613ea39167/jtnn/mpn.py#L26-L31

Just curious, do we consider to enrich the node features and edge features for ZINC dataset in `pytorch_geometric`?

**Reply by rusty1s** [reply]
I don't think we want to do that as the `ZINC` dataset should mimic the data used in the `benchmarking-gnns` repository to ensure comparability between reported results.

**Reply by guo00413** [reply]
Thanks and yes, to keep it same as benchmark is important! As planed, so I would build a converting method to build a complete graph data for ZINC.
